service/worker/scanner/tasklist/scavenger_test.go

```diff
@@ -55,6 +55,19 @@ const (
 
 var errTest = errors.New("transient error")
 
+type MockConfig struct {
```

Comment:
like above comment, if it's for testing, you can do similar as https://github.com/uber/cadence/blob/bfcb03fe53883ff4d574d9cc41e20dab202d4d52/service/worker/scanner/history/scavenger_test.go#L73

Reply:
The problem with this approach is it means scavenger needs to store a pointer to scanner.Config. That creates an import cycle, which Go does not allow. Using an interface here breaks the loop.

Reply:
>it means scavenger needs to store a pointer to scanner.Config

this is not what I meant. You can put these three fields here, similar to other fields: https://github.com/uber/cadence/blob/master/service/worker/scanner/scanner.go#L69

Reply:
They are already there. The problem is scavenger cannot reference scanner.Config in order to access them because that requires importing scanner - but scanner already imports scavenger. That makes a loop and is not allowed.

Reply:
nm I think I see what you mean